src/middlewared/middlewared/alembic/versions/23.10/2023-03-24_22-57_2fa_users.py

```diff
@@ -0,0 +1,54 @@
+"""
+2FA support for multiple usrers
+
+Revision ID: 55836e7dac39
+Revises: c55b034b7654
+Create Date: 2023-03-24 22:57:01.757983+00:00
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+revision = '55836e7dac39'
+down_revision = 'c55b034b7654'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # We want 3 things here:
+    # 1) Have a model where we can keep user 2fa data
+    # 2) Have a model where global 2fa configurations can be enforced
+    # 3) Migrate
+
+    op.create_table(
+        'account_twofactor_user_auth',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('secret', sa.String(length=16), nullable=True),
+        sa.Column('user_id', sa.Integer(), nullable=True),
+        sa.ForeignKeyConstraint(
```

Comment:
How will this work with users from directory services? They won't be present in our sqlite database.

Reply:
They would be handled in a subsequent PR - these changes only account for local users

Reply:
Even if you're not implementing some AD / LDAP wrapper now, we shouldn't make implementation decisions that prevent it in the future.

Reply:
Sure, the idea is that when we add supoprt for them we would be adding another column for a unique id which will point to the AD user and we will be making appropriate changes then. So the current workflow is handled in a way so we can account for AD users in the future.